pbs/pbsresponse.go

```diff
@@ -1,26 +1,38 @@
 package pbs
 
+// PBSBid is a bid from the auction. These are produced by Adapters, and target a particular Ad Unit.
 type PBSBid struct {
-	BidID             string            `json:"bid_id"`
-	AdUnitCode        string            `json:"code"`
-	Creative_id       string            `json:"creative_id,omitempty"`
-	BidderCode        string            `json:"bidder"`
-	BidHash           string            `json:"-"` // this is the hash of the bidder's unique bid identifier for blockchain. Should not be sent to browser.
-	Price             float64           `json:"price"`
-	Currency          string            `json:"currency,omitempty"`
-	NURL              string            `json:"nurl,omitempty"`
-	Adm               string            `json:"adm,omitempty"`
-	Width             uint64            `json:"width,omitempty"`
+	// BidID identifies the Bid Request within the Ad Unit which this Bid targets. It should match one of
+	// the values inside PBSRequest.AdUnits[i].Bids[j].BidID.
+	BidID string `json:"bid_id"`
+	// AdUnitCode identifies the AdUnit which this Bid targets.
+	// It should match one of PBSRequest.AdUnits[i].Code, where "i" matches the AdUnit used in
+	// as BidID.
+	AdUnitCode  string `json:"code"`
+	Creative_id string `json:"creative_id,omitempty"`
+	BidderCode  string `json:"bidder"`
+	// BidHash is the hash of the bidder's unique bid identifier for blockchain. It should not be sent to browser.
+	BidHash string `json:"-"`
+	// Price is the cpm, in US Dollars, which the bidder is willing to pay if this bid is chosen.
+	// TODO: Add support for other currencies someday.
+	Price float64 `json:"price"`
+	NURL  string  `json:"nurl,omitempty"`
+	// Adm is the payload which should be used to deliver the ad, if this bid is chosen.
+	Adm string `json:"adm,omitempty"`
+	// Width is the intended width which Adm should be shown, in pixels.
+	Width uint64 `json:"width,omitempty"`
+	// Height is the intended width which Adm should be shown, in pixels.
 	Height            uint64            `json:"height,omitempty"`
 	DealId            string            `json:"deal_id,omitempty"`
 	CacheID           string            `json:"cache_id,omitempty"`
 	ResponseTime      int               `json:"response_time_ms,omitempty"`
 	AdServerTargeting map[string]string `json:"ad_server_targeting,omitempty"`
 }
 
+// PBSBidSlice attaches the methods of sort.Interface to []PBSBid, ordering them by response times.
```

Comment:
aha... that makes far more sense. Totally missed that because of the code style (minus sign looked like a hyphen).

Looks like another issue here too... the `responseTime` is in `ms`, and the value is being divided by `1000000000`. That should make it "response time in terraseconds," and not "response time in nanoseconds." If it were really nanoseconds, then the bid time _would_ probably take precedence over the price when ordering.